src/clean_companies_data.py

```diff
@@ -0,0 +1,169 @@
+# Done by checking if first word in a category is present in other category labels.
+# Example 1: word "advertising" apppears in "advertising platforms".
+# Example 2: "enterprise" is the first word in "enterprise software" apppears in "enterprise 2.0"
+
+for key in tqdm(top_50_dict.keys()):
+    # Initialize each key with empty list
+    top_50_dict[key] = []
+    for cat in outside_50_markets:
+        # Check if first word is present in category, add to key value list, remove from category list.
+        if key.split(' ')[0] in cat.split(' '):
+            top_50_dict[key].append(cat)
+            outside_50_markets.remove(cat)
+        elif (len(key.split(' ')) > 1) and (key.split(' ')[1] == 'and') and (key.split(' ')[2] in cat.split(' ')):
+            top_50_dict[key].append(cat)
+            outside_50_markets.remove(cat)
+
+# Check how many categories are left.
+remainder_outside_50 = outside_50_markets
+len(remainder_outside_50)
+
+# Second pass at collapsing market categories.
+# Market category labels are split into character n-grams.
+# N-gram similarity Dice metric is calculated with intersection/union of n-gram sets.
+
+def word2ngrams(text, n=4, exact=True):
+    """Convert text into ngrams."""
+    return ["".join(g) for g in zip(*[text[i:] for i in range(n)])]
+
+def sim4gram(text1, text2):
+    """Calculate ngram similarity metric."""
+    list1 = word2ngrams(text1)
+    list2 = word2ngrams(text2)
+    return len(list(set(list1) & set(list2)))/len(list(set(list1) | set(list2)))
+
+for key in tqdm(top_50_dict.keys()):
+    for cat in remainder_outside_50:
+        if sim4gram(key, cat) > 0.25:
+            top_50_dict[key].append(cat)
+            remainder_outside_50.remove(cat)
+
+# Check how many categories are left.
+len(remainder_outside_50)
+
+# Assign remainder of categories to "Other"
+top_50_dict['other'] = remainder_outside_50
+
+def invert_dict(d):
+    """Function to invert a dictionary of lists so list values become keys."""
+    return dict( (v,k) for k in d for v in d[k] )
+
+category_replace_dict = invert_dict(top_50_dict)
+
+# Relabel key value pairs that do not match well.
+category_replace_dict['big data analytics'] = 'big data'
+category_replace_dict['coworking'] = 'other'
+category_replace_dict['mining technologies'] = 'other'
+category_replace_dict['mobility'] = 'other'
+category_replace_dict['nanotechnology'] = 'other'
+category_replace_dict['real time'] = 'other'
+category_replace_dict['space travel'] = 'other'
+category_replace_dict['social network media'] = 'social media'
+
+# Replace values in market column based on replacement dictionary.
+company_df['market'] = company_df['market'].map(category_replace_dict).fillna(company_df['market'])
+
+# One-hot encoding of startup status.
+company_df['status'].value_counts()
+
+company_df = pd.concat([company_df, pd.get_dummies(company_df['status'], prefix='status')], axis=1)
+company_df = company_df.drop('status', axis=1)
```

Comment:
could be preference, but recent pandas versions allow `df.drop(columns='status')` for more clarity (`axis=0,1` less readable)